### PR TITLE
Bumps go-oidc version to include fix for jwt header parsing

### PR DESCRIPTION
**Pull request by thomastaylor312**
**What this PR does / why we need it**:
This bumps the go-oidc dependency to use a fix merged in https://github.com/coreos/go-oidc/pull/153 for OIDC providers that don't set an `Expires` header

**Which issue this PR fixes** : 
Partially addresses #42654 
Also related: https://github.com/coreos/go-oidc/issues/136

**Special notes for your reviewer**:
None

**Release note**:
```release-note
NONE
```

### CODE REVIEW

**Comment by ericchiang** [comment]
See https://github.com/kubernetes/kubernetes/pull/45056 for expected comment format.

Thanks for this! Will ok to test after the comment update.

**Comment by thomastaylor312** [comment]
@ericchiang Thanks for the pointer on that. Commit message has been updated.

File: Godeps/Godeps.json [hunk 1]
```diff
@@ -832,12 +832,12 @@
 		},
 		{
 			"ImportPath": "github.com/docker/distribution/digest",
-			"Comment": "v2.4.0-rc.1-38-gcd27f17",
+			"Comment": "v2.4.0-rc.1-38-gcd27f179",
```
Comment (ericchiang) [inline Godeps/Godeps.json]: Any idea why this got bumped?

Comment (thomastaylor312) [inline Godeps/Godeps.json]: A lot of the comments got bumped. Not the actual versions, but it added one extra digit of the git hash on there. I can remove if needed

**Comment by brendandburns** [comment]
@thomastaylor312 looks like you need to run `./hack/update-staging-client-go.sh`

**Comment by thomastaylor312** [comment]
Hmm...I thought had run that, but let me try again

**Comment by thomastaylor312** [comment]
```shell
$ ./hack/update-staging-client-go.sh
Checking whether godeps are restored
Running staging/copy.sh
  creating the tmp directory
  copying client packages
  rewriting imports
  rearranging directory layout
  running gofmt
  remove black listed files
  move to the client repo
$ git status
On branch bump_oidc_dep
Your branch is up-to-date with 'thomastaylor312/bump_oidc_dep'.
nothing to commit, working tree clean
```
Everything seems to be in order, but when I run `./hack/verify-staging-godeps.sh`, it has the same error as the test. Am I missing something?

**Comment by ericchiang** [comment]
@thomastaylor312 I think `./hack/update-staging-client-go.sh` acts on your GOPATH. You might need to run `./hack/godep-restore.sh` or make sure the go-oidc in your GOPATH is the correct, new version.

**Comment by thomastaylor312** [comment]
I just pushed what I think should fix it. I had to run `./hack/update-staging-godeps.sh`

**Comment by brendandburns** [comment]
@thomastaylor312 looks like you need to sign the CLA.

**Comment by thomastaylor312** [comment]
I definitely have the CLA signed. Let me check what is happening

**Comment by thomastaylor312** [comment]
Signed it!

**Comment by thomastaylor312** [comment]
I signed it!

**Comment by thomastaylor312** [comment]
I don't know why it isn't checking for the CLA again, but I double checked and did sign it

**Comment by ericchiang** [comment]
@thomastaylor312 make sure you've signed it with the same email that your git commit is composed with.

**Comment by thomastaylor312** [comment]
@ericchiang I did do that. Both commits have my email on there and are the email associated with the CLA. It was good when I opened the PR, so I am not quite sure what happened

**Comment by thomastaylor312** [comment]
Re-signed the CLA

**Comment by gtaylor** [comment]
Is this going to end up in Kubernetes 1.7.4+, or will we need to wait for 1.8? We too are being hit by this in trying to use our Okta IDP.

**Comment by ericchiang** [comment]
@gtaylor we've seen other issues with using Okta as an IdP https://github.com/kubernetes/kubernetes/issues/46806

Can you take a look at that and make sure you're not impacted?

**Comment by gtaylor** [comment]
@ericchiang For the time being, Okta is our source of truth, so everyone ends up with verified emails. We're getting this, which appears to be the cache header 0-value issue:
```
E0807 22:47:22.056422       1 oidc.go:178] oidc authenticator: failed to fetch provider discovery data: parsing time "0" as "Mon, 02 Jan 2006 15:04:05 MST": cannot parse "0" as "Mon"
E0807 22:47:22.056446       1 authentication.go:58] Unable to authenticate the request due to an error: [invalid bearer token, [crypto/rsa: verification error, invalid bearer token, fetch provider config: parsing time "0" as "Mon, 02 Jan 2006 15:04:05 MST": cannot parse "0" as "Mon"]]
```

**Comment by thomastaylor312** [comment]
@gtaylor That definitely is the header issue

**Comment by gtaylor** [comment]
@thomastaylor312 while you were going through this review process, did you get any indication of when this might make it into a release? I'm trying to figure out how to proceed with our Kubernetes rollout.

**Comment by ericchiang** [comment]
@gtaylor I'll ask about a cherry-pick during tomorrow's SIG auth meeting.

**Comment by gtaylor** [comment]
@ericchiang That would be awesome. I can try to attend this, depending on whether BART cooperates.

**Comment by ericchiang** [comment]
@gtaylor @thomastaylor312 this was approved for a cherry pick by SIG auth. Can someone open one? If not I can do it.

**Comment by gtaylor** [comment]
Big thanks, @ericchiang and all. This is great news!

**Comment by thomastaylor312** [comment]
@ericchiang I can do it tomorrow. I need to learn how to do it anyway

**Comment by gtaylor** [comment]
Is this probably too late for 1.7.4?

**Comment by thomastaylor312** [comment]
@ericchiang I opened the PR, but this one needs the appropriate labels
